src/Command/ScanForErrorMiddleware/NextInvocationScanner.php

```diff
@@ -0,0 +1,120 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-expressive for the canonical source repository
+ * @copyright Copyright (c) 2016 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   https://github.com/zendframework/zend-expressive/blob/master/LICENSE.md New BSD License
+ */
+
+namespace Zend\Expressive\Command\ScanForErrorMiddleware;
+
+class NextInvocationScanner
+{
+    const TEMPLATE = <<< 'EOT'
+<?php
+%s
+EOT;
+
+    /**
+     * @var array
+     */
+    private $tokens;
+
+    /**
+     * @param string $code Code to tokenize and scan.
+     */
+    public function __construct($code)
+    {
+        $this->tokens = token_get_all(sprintf(
+            self::TEMPLATE,
+            $code
+        ));
+    }
+
+    /**
+     * Scan for invocations of $next() with error arguments.
+     */
+    public function scan()
+    {
+        $next                 = false;
+        $inNext               = false;
+        $parenDepth           = 0;
+        $commaCount           = 0;
+        $errorInvocationFound = false;
+
+        foreach ($this->tokens as $token) {
+            if (! $next && $this->isNext($token)) {
+                // Hit a $next variable
+                $next = true;
+                continue;
+            }
+
+            if ($next && ! $inNext && is_array($token)) {
+                // $next followed by token
+                $next = false;
+                continue;
+            }
+
+            if ($next && ! $inNext && $token === '(') {
+                // $next followed by opening paren (invocation)
+                $inNext = true;
+                $parenDepth = 1;
+                continue;
+            }
+
+            if (! $inNext) {
+                // Not in a $next invocation; nothing to examine
+                continue;
+            }
+
+            if (is_array($token)) {
+                // Hit a token; continue to next
+                continue;
+            }
+
+            if ($token === '(') {
+                // Hit an opening paren; increase paren depth
+                $parenDepth += 1;
```

Comment:
I've been using `+=` and `-=` for a few years, since seeing a Douglas Crockford talk where he demonstrated how it's more optimized both for the language interpreter (while he was talking about JavaScript, the optimizations are present for most C-style languages), as well as human comprehension (as `++` and `--` vary behavior based on whether or not they prefix or suffix the variable). Using the `+=` and `-=` syntax makes it clear that we are _re-assigning_ the value following the increment/decrement operation.

It essentially comes down to author preference; I've not been changing them during code reviews unless doing so aids in understanding how the code works.